File: LunamaroUi/LounamaroUI/src/app/Models/user-order-header.ts
import { PaymentType } from "./Admin/PaymentType";
import { OrderItem } from "./order-item";
import { OrderStatus } from "./orderStatus";

export interface UserOrderHeader {
  id: number;
  userId?: string | null;
  dateOfOrder: string;                 // DateTime → string in JSON
  dateOfShipping?: string | null;

  // 💰 Financial breakdown
  originalTotalAmount: number;         // before any discount
  offerDiscountAmount: number;         // product offers
  tierDiscountAmount: number;          // tier/global discount
  totalDiscountAmount: number;         // offer + tier
  finalTotalAmount: number;            // final price to pay
  freeProductId?: number | null;       // optional free product

  // 💳 Payment info
  stripeSessionId?: string | null;
  stripePaymentIntentId?: string | null;
  paymentStatus?: string | null;
  paymentProcessDate?: string | null;
  transactionId?: string | null;
  paymentType: PaymentType;

  // 📦 Shipping info
  phoneNumber: string;
  deliveryStreetAddress: string;
  city: string;
  state: string;
  postalCode: number;
  name: string;
  trackingNumber?: string | null;
  carrier?: string | null;

  // 🔹 Status & items
  orderStatus: OrderStatus;            // enum in TS
  orderItems?: OrderItem[];            // list of items
}

File: LunamaroApi/Lunamaroapi/Models/UserOrderHeader.cs
using Lunamaroapi.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class UserOrderHeader
{
    public int Id { get; set; }

    public string TemporaryKey { get; set; } = null!;

    public string? UserId { get; set; }

    [ForeignKey("UserId")]
    public ApplicationUser? User { get; set; }

    [Required]
    public DateTime DateOfOrder { get; set; }

    public DateTime? DateOfShipping { get; set; }

    // 💰 Financial Breakdown

    [Required]
    public decimal OriginalTotalAmount { get; set; }      // Before ANY discount

    public decimal OfferDiscountAmount { get; set; }      // Product offers

    public decimal TierDiscountAmount { get; set; }       // Tier/global discount

    public decimal TotalDiscountAmount { get; set; }      // Offer + Tier

    [Required]
    public decimal FinalTotalAmount { get; set; }         // Final price user pays


    // 💳 Payment
    public string? StripeSessionId { get; set; }
    public string? StripePaymentIntentId { get; set; }
    public string? PaymentStatus { get; set; }
    public DateTime? PaymentProcessDate { get; set; }
    public string? TransactionId { get; set; }

    // 📦 Shipping Info
    [Required]
    public string PhoneNumber { get; set; } = null!;

    [Required]
    public string DeliveryStreetAddress { get; set; } = null!;

    [Required]
    public string City { get; set; } = null!;

    [Required]
    public string State { get; set; } = null!;

    [Required]
    public int PostalCode { get; set; }

    [Required]
    public string Name { get; set; } = null!;

    public string? TrackingNumber { get; set; }
    public string? Carrier { get; set; }

    public PaymentType PaymentType { get; set; }

    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}